kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala

```diff
@@ -56,7 +56,7 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
 
   override def initialize(conf: KyuubiConf): Unit = synchronized {
     val host = conf.get(FRONTEND_REST_BIND_HOST)
-      .getOrElse(Utils.findLocalInetAddress.getHostAddress)
+      .getOrElse(Utils.findLocalInetAddress.getCanonicalHostName)
```

Comment:
shall we respect the config `FRONTEND_CONNECTION_URL_USE_HOSTNAME` ? and it seems `getHostName` is enough since the StatefulSet ensures the each pod hostname.

Reply:
no,  I deploy kyuubi in multiple k8s clusters.

It might be:
kyuubidev-0.35.k8s.io
kyuubidev-0.28.k8s.io

Reply:
> getHostName

Seems `getHostName` is a bug for me...

Reply:
shall we modify the `getHostName` to `getCanonicalHostName`? @ulysses-you

Reply:
`TFrontendService` should have a similar problem.
We encountered this problem when upgrading 1.5 and changed to `getCanonicalHostName`
I see that `KyuubiMySQLFrontendService` still uses `getCanonicalHostName`.

Reply:
> We encountered this problem when upgrading 1.5 and changed to getCanonicalHostName

I also meet that issue and I set `kyuubi.engine.connection.url.use.hostname=false` manually.

> shall we respect the config FRONTEND_CONNECTION_URL_USE_HOSTNAME
I think that we should always use CanonicalHostName for the kyuubiInstance url in metastore.

Because I set `kyuubi.engine.connection.url.use.hostname=false` before, if I respect FRONTEND_CONNECTION_URL_USE_HOSTNAME, I will use ipaddress for kyuubiInstance.

Reply:
the connection url for rest fe is special.